lib/marqeta/transaction.rb

```diff
@@ -60,5 +86,14 @@ def accessible_time_attributes
     def card_acceptor_hash
       attributes_hash['card_acceptor']
     end
+
+    def gateway_response_code
+      gpa_order = attributes_hash['gpa_order']
+      gpa_order['funding']['gateway_log']['response']['code'] unless gpa_order.nil?
```

Comment:
Probably worth using `fetch` here – it gives a more useful error message if something in the hash isn't as expected.

Reply:
@andyw8 👍